hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -250,42 +250,36 @@ public void handleRound(
                 migrationStateChanges.clear();
             }
         }
-        final boolean isGenesis =
-                switch (streamMode) {
-                    case RECORDS -> blockRecordManager
-                            .consTimeOfLastHandledTxn()
-                            .equals(Instant.EPOCH);
-                    case BLOCKS, BOTH -> blockStreamManager.pendingWork() == GENESIS_WORK;
-                };
-        boolean systemTransactionsDispatched = false;
-        if (isGenesis) {
-            final var genesisEventTime = round.iterator().next().getConsensusTimestamp();
-            logger.info("Doing genesis setup before {}", genesisEventTime);
-            systemTransactions.doGenesisSetup(genesisEventTime, state);
-            systemTransactionsDispatched = true;
-            if (streamMode != RECORDS) {
-                blockStreamManager.confirmPendingWorkFinished();
-            }
-            logger.info(SYSTEM_ENTITIES_CREATED_MSG);
-            requireNonNull(systemEntitiesCreatedFlag).set(true);
-        }
-
+        systemTransactions.resetNextDispatchNonce();
         recordCache.resetRoundReceipts();
+        boolean transactionsDispatched = false;
         try {
-            handleEvents(state, round, systemTransactionsDispatched, stateSignatureTxnCallback);
+            transactionsDispatched |= handleEvents(state, round, stateSignatureTxnCallback);
+            try {
+                transactionsDispatched |= nodeRewardManager.maybeRewardActiveNodes(
+                        state,
+                        boundaryStateChangeListener.lastConsensusTimeOrThrow().plusNanos(1),
+                        systemTransactions);
+            } catch (Exception e) {
+                logger.warn("Failed to reward active nodes", e);
+            }
+            // Inform the BlockRecordManager that the round is complete, so it can update running hashes in state
+            // from results computed in background threads. The running hash has to be included in state, but we want
+            // to synchronize with background threads as infrequently as possible; per round is the best we can do
```

Comment:
Which background threads are we talking about here?

Reply:
If we look at how these running hashes are computed, they are done on separate asynchronous threads that are concurrently computing parts of the running hash. When the round ends, calling `blockRecordManager.endRound(state)` synchronizes the results from these background computations.

Reply:
Ah the hashing threads, got it, thanks!